sdk/servicebus/service-bus/src/serviceBusAtomManagementClient.ts

```diff
@@ -0,0 +1,547 @@
+
+import { QueueResourceSerializer } from "./serializers/queueResourceSerializer";
+import { TopicResourceSerializer } from "./serializers/topicResourceSerializer";
+import { SubscriptionResourceSerializer } from "./serializers/subscriptionResourceSerializer";
+import { RuleResourceSerializer } from "./serializers/ruleResourceSerializer";
+
+/**
+ * All operations return a `Promise<HttpOperationResponse>` object of which the `parsedBody`
+ * is populated with the response in JSON format with following structure:
+ *  {
+ *   response: <actual response>,
+ *   result: <parsed result, if any>,
+ *   error: <error information, if any>
+ *  }
+ */
+export class ServiceBusAtomManagementClient extends ServiceClient {
+  endpoint: any;
+
+  queueResourceSerializer: ResourceSerializer;
+  topicResourceSerializer: ResourceSerializer;
+  subscriptionResourceSerializer: ResourceSerializer;
+  ruleResourceSerializer: ResourceSerializer;
+
+  /**
+   * Initializes a new instance of the ServiceBusManagementClient class.
+   * @param credentials Credentials needed for the client to connect to Azure.
+   * @param options The parameter options
+   */
+  constructor(connectionString: string, proxySettings?: ProxySettings) {
+    const connectionStringObj = ServiceBusAtomManagementClient.parseConnectionString(
+      connectionString
+    );
+    const credentials = new SasServiceClientCredentials(
+      connectionStringObj.SharedAccessKeyName,
+      connectionStringObj.SharedAccessKey
+    );
+
+    const requestPolicyFactories: RequestPolicyFactory[] = [];
+    requestPolicyFactories.push(userAgentPolicy());
+    requestPolicyFactories.push(logPolicy(httpAtomXml));
+    requestPolicyFactories.push(atomSerializationPolicy());
+    requestPolicyFactories.push(signingPolicy(credentials));
+
+    if (proxySettings) {
+      requestPolicyFactories.push(proxyPolicy(proxySettings));
+    }
+    const serviceClientOptions: ServiceClientOptions = {
+      requestPolicyFactories: requestPolicyFactories
+    };
+
+    super(credentials, serviceClientOptions);
+    this.queueResourceSerializer = new QueueResourceSerializer();
+    this.topicResourceSerializer = new TopicResourceSerializer();
+    this.subscriptionResourceSerializer = new SubscriptionResourceSerializer();
+    this.ruleResourceSerializer = new RuleResourceSerializer();
+    this.endpoint = (connectionString.match("Endpoint=sb://(.*).servicebus.windows.net") || "")[1];
+  }
+
+  /**
+   * Creates a queue.
+   *
+   * @param {string}             queuePath                                             A string object that represents the name of the queue to create.
+   * @param {object}             [options]                                             The request options.
+   * @param {int}                [options.MaxSizeInMegaBytes]                          Specifies the maximum queue size in megabytes. Any attempt to enqueue a message that will cause the queue to exceed this value will fail.
```

Comment:
The docs for all the options are appearing as siblings to the queuePath and options arguments giving the impression that `options.MaxSizeInMegaBytes`, `options.DefaultMessageTimeToLive` etc are a ll individual parameters that can be passed to the `createQueue()` method. The docs here need to be re-visited.